### PR TITLE
#162948930 Like and Dislike Comments

### PR DESCRIPTION
**Pull request by danuluma**
#### What does this PR do?
Add ability to like and dislike comments

#### Description of Task(s) to be completed?
* Users should be able to like and or dislike comments

#### How should this be manually tested?
* Pull the branch `git fetch origin ft-like-comment-162948930:ft-like-comment-162948930`
* Checkout to the branch `git checkout ft-like-comment-162948930`
* Install requirements `pip3 install -r requirements.txt`
* Run the server `python3 manage.py runserver`
* Launch postman and create a user by sending a POST request to `http://127.0.0.1:8000/api/v1/users/` with a body of:
```
{
   "user":{
        "username": "username",
        "email": "valid@email.com",
        "password": "Passw0rd"
    }
}
```
* Login by sending a POST request to `http://127.0.0.1:8000/api/v1/users/login/` with your registered email and password
```
{
   "user":{
        "email": "valid@email.com",
        "password": "Passw0rd"
    }
}
```
* Copy the returned JWT token and store it somewhere
* Use postman to create an article by sending a POST request to `http://127.0.0.1:8000/api/v1/articles`
* The body should have the format of 
```
{ 
  "article":{
       "image_path":"dan",
            "title":"rick",
            "description":"and",
            "body":"morty"
    }
}
```
* Select Bearer Token under Authorization and paste in the token gotten previously.
* The response should be similar to 
```
{
    "id": 14,
    "author": {
        "email": "ulumadan2@gmail.com",
        "username": "newuser"
    },
    "image_path": "dan",
    "slug": "rick-5",
    "title": "rick",
    "description": "and",
    "body": "morty",
    "created_at": "2019-01-23T14:51:16.512539Z",
    "updated_at": "2019-01-23T14:51:16.512733Z",
    "favourites": false
}
```
* Take note of the value of slug attribute returned in the response. We'll need it next.
* Lets now create a comment. Send a `POST` request to `http://127.0.0.1:8000/api/v1/articles/<slug>/comments` with a body of 
```
{"comment":{
	"body":"comment 2"
	}
}
```
* You should get a response similar to :
```
{
    "message": "Comment Successfully added",
    ...
    "id": 2,
    "like_count": 0,
    "dislike_count": 0,
    "like_status": "Null"
}
```
* Take note of the comment ID
* To like this comment, we need to send a post request to `http://127.0.0.1:8000/api/v1/articles/<slug>/comments/<ID>/like`. Replace the <slug> and <ID> params with the actual slug and ID.
* To dislike, we again send a POST request to `http://127.0.0.1:8000/api/v1/articles/<slug>/comments/<ID>/dislike`, replacing <slug> and <ID> with the actual slug and ID.
* The response should be similar to:
```
{
    "message": {
        "result": "Great, you've liked this",
        "status": "Liked",
        "like_count": 1,
        "dislike_count": 0
    }
}
```
* Now if we view that article, by sending a `GET` request to `http://127.0.0.1:8000/api/v1/articles/<slug>/comments/<ID>/`, we should receive a response similar to:
```
{
    "comment": {
         …
        "like_count": 1,
        "dislike_count": 0,
        "like_status": "Liked"
    }
}

```

#### Any background context you want to provide?
* This project uses the latest version of python3. Please download it from [here](https://www.python.org/downloads/)
* You may as well use pip instead of pip3 if you have either only python3 installed on your computer or you have it set as the default version of python
* I did some minor code refactor(improve dry-ness) and pep8 fixes as well, especially in article and authentication apps

#### What are the relevant pivotal tracker stories?
[#162948930](https://www.pivotaltracker.com/story/show/162948930)

### CODE REVIEW

File: authors/urls.py [hunk 1]
```diff
@@ -46,4 +48,7 @@
     path('api/v1/', include('authors.apps.profiles.urls'), name='apiv1'),
     path('api/v1/', include('authors.apps.articles.urls')),
     path('api/v1/', include('authors.apps.rating.urls')),
+    path('api/v1/', include('authors.apps.comments.urls')),
+    path('api/v1/', include('authors.apps.rating.urls')),
+    path('api/v1/', include('authors.apps.comments.urls')),
```
Comment (boltonotieno) [inline authors/urls.py]: You have two similar paths ```path('api/v1/', include('authors.apps.comments.urls'))```
 kindly remove one cause there the same

**Review by boltonotieno — CHANGES_REQUESTED** [review]

File: authors/urls.py (hunk 1 above)
Comment (danuluma) [reply authors/urls.py]: Thanks, I'm correcting that

File: authors/apps/like_dislike/views.py [hunk 2]
```diff
@@ -16,26 +16,39 @@
...
     swagger_schema = SwaggerAutoSchema
 
-    def post(self, request, slug):
-        article = Article.objects.get(slug=slug)
-        pref_status = None
+    @swagger_auto_schema(
+        request_body=swagger_body(None),
+        responses=status_codes(codes=(200, 401))
+    )
+    def post(self, request, slug, pk=None):
```
Comment (Ruiru11) [inline authors/apps/like_dislike/views.py]: add comments helps us understand what is happening

Comment (danuluma) [reply authors/apps/like_dislike/views.py]: Cool, thanks,,,let me do that

**Review by boltonotieno — APPROVED** [review]

**Pull request merged**